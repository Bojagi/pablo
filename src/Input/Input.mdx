---
name: Input
route: /input
menu: Components
---

import { Props, Playground } from 'docz'
import { Box } from '../Box/Box.tsx'
import { Input } from './Input.tsx'
import { IconButton } from '../IconButton/index.ts'
import { Eye, EyeOff, Search } from 'react-feather';

# Input

Form element that allows users to type in single line text. For multi-line text you can use the `TextArea` component.

## Examples

### Simple Input

The simplest form of an `Input` is just the field itself. You can pass a `useState` setter function to update state
as the first argument of `onChange` is the value itself. 
The second argument is the full `onChange` event of the native `input` element.

<Playground>
  {() => {
    const [value, setValue] = React.useState('');
    return (
      <Input value={value} onChange={setValue} />
    );
  }}
</Playground>

### Input with label

You should always add context to an input field so users know what to put into the field. For this you can pass the 
`label` prop with the content for the `<label>` element (can be string or JSX).

<Playground>
  {() => {
    const [name, setName] = React.useState('');
    return (
      <Input label="Name" value={name} onChange={setName} />
    );
  }}
</Playground>

### Input with additional information

Additionally you can also pass the `infoText` prop to give hints on how to fill the field.

<Playground>
  {() => {
    const [name, setName] = React.useState('');
    return (
      <Input label="Name" infoText="Don't use a fake name" value={name} onChange={setName} />
    );
  }}
</Playground>

### Input with error

When you do validation on a field and the user input isn't valid, you can use the `error` prop and overwrite the `infoText` content with it.

<Playground>
  {() => {
    const [name, setName] = React.useState('');
    const hasError = React.useMemo(() => name.length && name !== 'Hans');
    return (
      <Input label="Name" error={hasError && "The only non fake name is Hans!"} infoText={"Don't use a fake name"} value={name} onChange={setName} />
    );
  }}
</Playground>

### Input with end or start component

Sometimes you want to make the input more interactive. For this you can use components passed to the `end` or `start` prop.

<Playground>
  {() => {
    const [password, setPassword] = React.useState('');
    const [showPassword, setShowPassword] = React.useState(false);
    return (
      <Input 
        type={showPassword ? 'text' : 'password'}
        label="Password" 
        value={password} 
        onChange={setPassword} 
        end={<IconButton size="small" onClick={() => setShowPassword(!showPassword)}>
          {showPassword ? <Eye /> : <EyeOff />}
        </IconButton>} 
      />
    );
  }}
</Playground>

You can also use it for an inlined button (use `placeholder` instead of `label` prop).

<Playground>
  {() => {
    const [searchTerm, setSearchTerm] = React.useState('');
    return (
      <Input 
        placeholder="Search"
        value={searchTerm} 
        onChange={setSearchTerm} 
        start={<IconButton ml={-2} size="small"><Search /></IconButton>} 
      />
    );
  }}
</Playground>

## Import

```js
import { Input } from '@bojagi/pablo/Input';
```

## Props
<Props of={Input} />
