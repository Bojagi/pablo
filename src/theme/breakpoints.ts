export type Breakpoint = 'sm' | 'md' | 'lg' | 'xl';

export const breakpointNames: Breakpoint[] = ['sm', 'md', 'lg', 'xl'];

export type BreakpointsArray = [number, number, number, number];

export type Breakpoints = BreakpointsArray &
  Record<Breakpoint, number> & {
    breakpointNames: Breakpoint[];
  };

export function createBreakpoints(
  array: BreakpointsArray,
  usedBreakpointNames: Breakpoint[] = breakpointNames
): Breakpoints {
  const newBreakpoints = ([...array] as unknown) as Breakpoints;
  usedBreakpointNames.forEach((bp, index) => {
    newBreakpoints[bp] = newBreakpoints[index];
  });
  newBreakpoints.breakpointNames = usedBreakpointNames;

  return newBreakpoints as Breakpoints;
}

export const defaultBreakpoints: BreakpointsArray = [600, 1000, 1200, 1920];

export const breakpoints = createBreakpoints(defaultBreakpoints, breakpointNames);
