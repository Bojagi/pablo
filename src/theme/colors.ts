export type ShadedColor = {
  main: string;
  dark: string;
  darkest: string;
  light: string;
  lightest: string;
  contrastText: string;
  contrastTextLight: string;
};

export type CommonColors = {
  white: string;
  black: string;
  whiteContrastText: string;
  blackContrastText: string;
};

export type TextColors = {
  main: string;
  info: string;
};

export type BorderColors = {
  main: string;
  lightest: string;
};

export interface GrayColors {
  '50': string;
  '100': string;
  '200': string;
  '300': string;
  '400': string;
  '500': string;
  '600': string;
  '700': string;
  '800': string;
  '900': string;
}

export type AllColors = ShadedColor & TextColors & BorderColors & CommonColors;

export interface Colors {
  common: CommonColors;
  gray: GrayColors;
  borders: BorderColors;
  brand: ShadedColor;
  positive: ShadedColor;
  negative: ShadedColor;
  neutral: ShadedColor;
  text: TextColors;
}

export const WHITE = '#ffffff';
export const BLACK = '#000000';

export const colors: Colors = {
  common: {
    white: WHITE,
    black: BLACK,
    whiteContrastText: BLACK,
    blackContrastText: WHITE,
  },
  gray: {
    '50': getGrayColor(0.95),
    '100': getGrayColor(0.9),
    '200': getGrayColor(0.8),
    '300': getGrayColor(0.7),
    '400': getGrayColor(0.6),
    '500': getGrayColor(0.5),
    '600': getGrayColor(0.4),
    '700': getGrayColor(0.3),
    '800': getGrayColor(0.2),
    '900': getGrayColor(0.1),
  },
  borders: {
    main: `${BLACK}${getHexByte(0.25)}`,
    lightest: `${BLACK}${getHexByte(0.1)}`,
  },
  text: {
    main: BLACK,
    info: `${BLACK}${getHexByte(0.5)}`,
  },
  brand: {
    lightest: '#EBE7FF',
    light: '#D6CEFF',
    main: '#6A50F2',
    dark: '#4129BD',
    darkest: '#281683',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  positive: {
    lightest: '#DFF6E6',
    light: '#BBECCB',
    main: '#38C765',
    dark: '#2FB85A',
    darkest: '#108B36',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  negative: {
    lightest: '#FFE7E5',
    light: '#FFCFCC',
    main: '#F21D0D',
    dark: '#CA271B',
    darkest: '#871A12',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  neutral: {
    lightest: '#FCF5DB',
    light: '#FFEB9E',
    main: '#FFDD5B',
    dark: '#EDC52A',
    darkest: '#D6AC0B',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
};

/**
 * Get hex color byte (e.g. to get transparency)
 * @param value Percentage value (between 0.0 and 1.0)
 */
function getHexByte(value: number) {
  return (Math.round((255 * value) / 16) * 16).toString(16);
}

function getGrayColor(value: number) {
  return `#${getHexByte(value)}${getHexByte(value)}${getHexByte(value)}`;
}
