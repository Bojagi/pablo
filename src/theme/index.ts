export * from './defaultTheme';
export * from './context';
export * from './PabloThemeProvider';
export * from './useComponentStyle';
