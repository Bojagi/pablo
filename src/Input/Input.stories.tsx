import React, { useState } from 'react';

import { Search } from 'react-feather';
import { css } from '@emotion/react';
import { Box, Flex } from '../Box';
import { IconButton } from '../IconButton';
import { Typography } from '../Typography';
import { Input, InputProps } from './Input';

export default {
  title: 'Input',
};

const ControlledInput = ({ value: valueInitial, ...props }: Omit<InputProps, 'onChange'>) => {
  const [value, setValue] = useState(valueInitial);
  return <Input value={value} mb={4} {...props} onChange={setValue} />;
};

const baseStory = (args) => <ControlledInput {...args} />;

export const FullWidth = baseStory.bind(null);
FullWidth.args = { fullWidth: true };

export const FixedWidth = baseStory.bind(null);
FixedWidth.args = { width: 800 };

export const Inline = baseStory.bind(null);
Inline.args = {};

export const WithLabel = baseStory.bind(null);
WithLabel.args = {
  label: <>name</>,
};

export const WithInfoText = baseStory.bind(null);
WithInfoText.args = {
  infoText: (
    <>
      This is something <strong>important!</strong>
    </>
  ),
};

export const WithLabelAndInfoText = baseStory.bind(null);
WithLabelAndInfoText.args = {
  label: <>name</>,
  infoText: (
    <>
      This is something <strong>important!</strong>
    </>
  ),
};

export const Error = baseStory.bind(null);
Error.args = {
  error: 'Something terrible happened!',
  infoText: (
    <>
      This is something <strong>important!</strong>
    </>
  ),
};

export const MultipleInputs = () => (
  <Box>
    <Box mb={5}>
      <ControlledInput label="first name" />
    </Box>
    <Box mb={5}>
      <ControlledInput label="last name" />
    </Box>
    <Box mb={5}>
      <ControlledInput label="credit card no." infoText="optional" />
    </Box>
  </Box>
);

export const MultipleInputsWithFlex = () => (
  <Flex width={300} gap={2} equal p={2}>
    <ControlledInput fullWidth label="name" />
    <ControlledInput
      fullWidth
      label="search"
      end={
        <IconButton size="small">
          <Search />
        </IconButton>
      }
    />
  </Flex>
);

export const WithEndComponent = baseStory.bind(null);
WithEndComponent.args = {
  label: <>search</>,
  fullWidth: false,
  width: 600,
  end: (
    <IconButton size="small">
      <Search />
    </IconButton>
  ),
};

export const WithStartComponent = baseStory.bind(null);
WithStartComponent.args = {
  fullWidth: false,
  width: 600,
  start: (
    <Typography inline variant="body" bold>
      search:
    </Typography>
  ),
};

export const Outline = baseStory.bind(null);
Outline.args = { variant: 'outline' };

export const WithCustomStyles = baseStory.bind(null);
WithCustomStyles.args = {
  label: <>Some longer name</>,
  infoText: (
    <>
      This is something <strong>important!</strong>
    </>
  ),
  customStyles: {
    label: css`
      transform: translateY(-36px) rotate(6deg);
      /* transform-origin: 0 0; */
      text-align: right;
    `,
    infoText: css`
      transform: rotate(-3deg);
      transform-origin: 0 0;
    `,
    wrapper: css`
      transform: rotate(-5deg);
      transform-origin: 0 0;
      background-color: blue;
    `,
    field: css`
      color: red;
    `,
  },
};
