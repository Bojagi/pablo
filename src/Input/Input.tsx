import React from 'react';
import styled from 'styled-components';
import { BoxProps } from '../Box';
import { getComponentStyle } from '../utils/styleHelpers/getComponentStyle';
import { BaseInput, InnerInputProps, InputVariant } from '../BaseInput/BaseInput';
import { useComponentStyle } from '../theme';

export interface InputProps extends BoxProps {
  id?: string;
  value?: string | number;
  error?: React.ReactNode;
  label?: React.ReactNode;
  variant?: InputVariant;
  infoText?: React.ReactNode;
  fullWidth?: boolean;
  width?: string | number;
  end?: React.ReactNode;
  onChange: (newValue: string, e: React.FormEvent<HTMLInputElement>) => void;
}

const InnerInput = styled.input<InnerInputProps>`
  box-sizing: border-box;
  flex-grow: 1;
  border: 0;
  padding: ${getComponentStyle('input.padding')};
  background-color: transparent;
  font-family: ${getComponentStyle('input.fontFamily')};
  outline: none;
`;

export function Input({ width, variant = 'filled', ...props }: InputProps) {
  const defaultWidth = useComponentStyle('input.defaultWidth');
  const adornmentGap = useComponentStyle('input.adornmentGap');
  return (
    <BaseInput
      name="input"
      width={width || defaultWidth}
      variant={variant}
      adornmentGap={adornmentGap}
      inputComponent={InnerInput}
      {...props}
    />
  );
}
