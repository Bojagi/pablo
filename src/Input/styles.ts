import { css } from 'styled-components';
import { getColor } from '../utils/styleHelpers/getColor';
import { getSpacing } from '../utils/styleHelpers/getSpacing';
import { Style } from '../theme/types';

export interface InputStyles {
  padding: Style;
  borderWidth: number;
  borderColor: Style;
  backgroundColor: Style;
  fontFamily: Style;
  transitions: string[][];
  focus: {
    outlineSize: Style;
    outlineColor: Style;
  };
  error: {
    borderColor: Style;
    focus: {
      outlineColor: Style;
    };
  };
}

export const inputStyles: InputStyles = {
  padding: css`
    ${getSpacing(0.75)} ${getSpacing(1.5)}
  `,
  fontFamily: (props) => props.theme.typography.base.fontFamily,
  borderWidth: 1,
  borderColor: getColor('borders'),
  backgroundColor: getColor('common', 'white'),
  focus: {
    outlineSize: getSpacing(0.375),
    outlineColor: getColor('brand', 'light'),
  },
  transitions: [
    ['border-color', '0.3s'],
    ['box-shadow', '0.3s', 'ease-in-out'],
  ],
  error: {
    borderColor: getColor('negative'),
    focus: {
      outlineColor: getColor('negative', 'light'),
    },
  },
};
