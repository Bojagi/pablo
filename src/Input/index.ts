import './styles';

export * from './Input';
