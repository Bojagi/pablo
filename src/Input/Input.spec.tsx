import { act, fireEvent, render } from '@testing-library/react';
import React from 'react';
import { PabloThemeProvider } from '../theme';
import { Input } from './Input';

test('Render without value', () => {
  const { container, getByTestId } = renderComponent({});
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-input')).toHaveValue('');
});

test('Render with value', () => {
  const { getByTestId } = renderComponent({ value: 'some' });
  expect(getByTestId('pbl-input')).toHaveValue('some');
});

test('Show label when available', () => {
  const { container, getByTestId } = renderComponent({
    value: 'some',
    label: (
      <>
        Your <strong>name</strong>
      </>
    ),
  });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-input-label')).toHaveTextContent('Your name');
});

test('Show infotext when available', () => {
  const { container, getByTestId } = renderComponent({
    value: 'some',
    infoText: (
      <>
        Something that <strong>helps</strong>
      </>
    ),
  });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-input-infotext')).toHaveTextContent('Something that helps');
});

test('Show error when an error message is given', () => {
  const { container, getByTestId } = renderComponent({
    value: 'some',
    error: (
      <>
        Something <strong>terrible</strong> happened!
      </>
    ),
  });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-input-infotext')).toHaveTextContent('Something terrible happened!');
});

test('Show error instead of infotext when both are set', () => {
  const { getByTestId } = renderComponent({
    value: 'some',
    error: (
      <>
        Something <strong>terrible</strong> happened!
      </>
    ),
    infoText: (
      <>
        something that <strong>helps</strong>
      </>
    ),
  });
  expect(getByTestId('pbl-input-infotext')).toHaveTextContent('Something terrible happened!');
});

test('Render with custom width when "width" prop is set', () => {
  const { getByTestId } = renderComponent({
    width: '600em',
  });

  expect(getByTestId('pbl-input-wrapper')).toHaveStyleRule('width', '600em');
});

test('Render with custom px width when number "width" prop is set', () => {
  const { getByTestId } = renderComponent({
    width: 600,
  });

  expect(getByTestId('pbl-input-wrapper')).toHaveStyleRule('width', '600px');
});

test('Render with width auto when "fullWidth" prop is set', () => {
  const { getByTestId } = renderComponent({
    width: 600,
    fullWidth: true,
  });

  expect(getByTestId('pbl-input-wrapper')).toHaveStyleRule('width', 'auto');
});

test('Trigger onChange event when the content of the inner input changed', () => {
  const onChangeMock = jest.fn();
  const { getByTestId } = renderComponent({
    value: 'some',
    onChange: onChangeMock,
  });

  expect(onChangeMock).toHaveBeenCalledTimes(0);
  act(() => {
    fireEvent.change(getByTestId('pbl-input'), { target: { value: 'newValue' } });
  });
  expect(onChangeMock).toHaveBeenCalledTimes(1);
  expect(onChangeMock).toHaveBeenCalledWith(
    'newValue',
    expect.objectContaining({ type: 'change' })
  );
});

test('Use generated id when no "id" prop is set', () => {
  const onChangeMock = jest.fn();
  const { getByTestId } = renderComponent({
    value: 'some',
    id: undefined,
    onChange: onChangeMock,
  });

  expect(getByTestId('pbl-input')).toHaveAttribute('id', expect.stringMatching(/:.*?:/));
});

function renderComponent(props) {
  return render(
    <PabloThemeProvider>
      <Input id="custom-id" {...props} />
    </PabloThemeProvider>
  );
}
