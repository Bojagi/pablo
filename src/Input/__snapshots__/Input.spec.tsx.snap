// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render without value 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 1px solid #00000040;
  border-radius: 8px;
  background-color: #ffffff;
  -webkit-transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  width: 300px;
}

.c1 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border: 0;
  padding: 6px 10px;
  background-color: transparent;
  font-family: "IBM Plex Sans",sans-serif;
  outline: none;
}

<div>
  <div
    class=""
  >
    <div
      class="c0"
      data-testid="pbl-input-wrapper"
      name="input"
      width="300"
    >
      <input
        class="c1"
        data-testid="pbl-input"
        id="custom-id"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Show error when an error message is given 1`] = `
.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.75rem;
  line-height: 1.5em;
  margin-bottom: 0;
  margin-top: 4px;
  color: #F44336;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 1px solid #00000040;
  border-radius: 8px;
  background-color: #ffffff;
  -webkit-transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  border-color: #F44336;
  width: 300px;
}

.c0:focus {
  box-shadow: 0 0 0 3px #FFD3D0;
}

.c1 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border: 0;
  padding: 6px 10px;
  background-color: transparent;
  font-family: "IBM Plex Sans",sans-serif;
  outline: none;
}

<div>
  <div
    class=""
  >
    <div
      class="c0"
      data-testid="pbl-input-wrapper"
      name="input"
      width="300"
    >
      <input
        class="c1"
        data-testid="pbl-input"
        id="custom-id"
        value="some"
      />
    </div>
    <p
      class="c2"
      color="negative.main"
      data-testid="pbl-input-infotext"
    >
      Something 
      <strong>
        terrible
      </strong>
       happened!
    </p>
  </div>
</div>
`;

exports[`Show infotext when available 1`] = `
.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.75rem;
  line-height: 1.5em;
  margin-bottom: 0;
  margin-top: 4px;
  color: #00000080;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 1px solid #00000040;
  border-radius: 8px;
  background-color: #ffffff;
  -webkit-transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  width: 300px;
}

.c1 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border: 0;
  padding: 6px 10px;
  background-color: transparent;
  font-family: "IBM Plex Sans",sans-serif;
  outline: none;
}

<div>
  <div
    class=""
  >
    <div
      class="c0"
      data-testid="pbl-input-wrapper"
      name="input"
      width="300"
    >
      <input
        class="c1"
        data-testid="pbl-input"
        id="custom-id"
        value="some"
      />
    </div>
    <p
      class="c2"
      color="text.info"
      data-testid="pbl-input-infotext"
    >
      Something that 
      <strong>
        helps
      </strong>
    </p>
  </div>
</div>
`;

exports[`Show label when available 1`] = `
.c0 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.4em;
  font-weight: 500;
  margin-bottom: 0.75em;
  margin-bottom: 6px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 1px solid #00000040;
  border-radius: 8px;
  background-color: #ffffff;
  -webkit-transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  width: 300px;
}

.c2 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border: 0;
  padding: 6px 10px;
  background-color: transparent;
  font-family: "IBM Plex Sans",sans-serif;
  outline: none;
}

<div>
  <div
    class=""
  >
    <label
      data-testid="pbl-input-label"
      for="custom-id"
    >
      <p
        class="c0"
      >
        Your 
        <strong>
          name
        </strong>
      </p>
    </label>
    <div
      class="c1"
      data-testid="pbl-input-wrapper"
      name="input"
      width="300"
    >
      <input
        class="c2"
        data-testid="pbl-input"
        id="custom-id"
        value="some"
      />
    </div>
  </div>
</div>
`;
